irc/server.go

```diff
@@ -462,6 +462,12 @@ func (server *Server) tryRegister(c *Client) {
 		return
 	}
 
+	// Check if connection requires SASL
+	if ConnectionRequiresSasl(c) {
```

Comment:
The division of responsibilities between this helper and its caller is a little awkward (one calls `Quit` and the other calls `destroy`). I feel like this would be easier to read if you pulled the whole helper back in here?